src/Engine/Objects/CombinedSkillValue.cpp

```diff
@@ -49,3 +51,31 @@ bool CombinedSkillValue::isMasteryValid(PLAYER_SKILL_MASTERY mastery) {
     return mastery == PLAYER_SKILL_MASTERY_NONE || mastery == PLAYER_SKILL_MASTERY_NOVICE || mastery == PLAYER_SKILL_MASTERY_EXPERT ||
            mastery == PLAYER_SKILL_MASTERY_MASTER || mastery == PLAYER_SKILL_MASTERY_GRANDMASTER;
 }
+
+void CombinedSkillValue::learn() {
```

Comment:
I don't like this from the code readability perspective.

The idea here is that `learn()` should only be used on skills that are currently set to zero, but the method doesn't check for that.

I think a better design would be to add factory methods to `CombinedSkillValue`:

```
static CombinedSkillValue rookie() {
    return CombinedSkillValue(1, PLAYER_SKILL_MASTERY_NOVICE);
}
```

Then just use assignment to convey the purpose of what's going on:

```
pActiveSkills[PLAYER_SKILL_CLUB] = CombinedSkillValue::rookie(); // Hidden skills, always known.
pActiveSkills[PLAYER_SKILL_MISC] = CombinedSkillValue::rookie(); 
```

`rookie` is a random name I've come up with, you can think of a better one.